TROUBLESHOOTING.md

```diff
@@ -237,4 +237,33 @@ The, given the PID, it is possible to grab the stack traces with the following c
 ./scripts/dump_stacks.py PID > /tmp/stack-traces.log
 ```
 
+## Using debug_run.py To Collect Debug Information
 
+A utility is provided in `scripts/debug_run.py` which can be used to create a `tar.gz`
+archive with the information required to debug _PyTorch/XLA_ executions.
+
+Example:
+
+```Shell
+./scripts/debug_run.py --outfile /tmp/debug_run.tar.gz -- python -u SCRIPT [ARGS...]
+```
+
+The _python_ `-u` flag is suggested to disable buffering so that captured logs are correctly
+interleaved (otherwise STDOUT will be rendered after all STDERR).
+
+The above command line example will leave the temporary folder containing the archived
+information on the filesystem. Use the `--tidy` flag to have that removed on exit:
+
+```Shell
+./scripts/debug_run.py --tidy --outfile /tmp/debug_run.tar.gz -- python -u SCRIPT [ARGS...]
+```
+
+The `debug_run.tar.gz` file should then be attached to bug reports when necessary.
+
+Since the script will collect a lot of data, it should usually be let run for no more
+than hundred steps or so.
+If the SCRIPT has arguments to control the number of steps, those should be used,
+otherwise hitting `CTRL^C` will interrupt the run.
```

Comment:
can we use xla counters to kill, e.g. at step 100, without using SCRIPT at all?

Reply:
I have used it during the weekend, and I have found CTRL-C worked just fine.
Autokill might be doable, but adds a bit of complexity, and I find the user looking at the messages can do a better job.
Will see later, once we get used to the tool.